mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -971,6 +975,10 @@ class LogisticRegression @Since("1.2.0") (
     }
 
     if (fitWithMean) {
+      // orginal initialCoefWithInterceptArray is for problem:
```

Comment:
As I mentioned in my review, I think it is a bad idea to make this adjustment to the starting intercept.  If you look at the current code that creates the initial intercept: https://github.com/apache/spark/blob/master/mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala#L893-L907
the intercept is set to equal the log(odds) of the data.  This is actually the optimal starting intercept for centered data (and not a great starting point for non-centered data).  So doing the adjustment here is actually counter productive.

Reply:
the `initialCoefWithInterceptArray` is intialized in two ways:
1, all zero weights with intercept = `b = \log{P(1) / P(0)} = \log{count_1 / count_0}`, then this adjustment does nothing;
2, a user-provided model `val providedCoef = optInitialModel.get.coefficientMatrix`, in this case, we should adjust the initial intercept: suppose user set maxIter=1 or even 0, the final intercept well be adjusted to a wrong value. Disable this intialization adjust will make test case 'set initial model'  fail:
![image](https://user-images.githubusercontent.com/7322292/109908528-6b056900-7cdf-11eb-8fd8-d2a68fc6fcfc.png)

Reply:
ok, if you are allowing users to input a starting point, then this makes sense.